src/components/TimeZonePicker.vue

```diff
@@ -0,0 +1,148 @@
+<!-- ADD_LICENSE_HEADER -->
+<template>
+	<div class="input-group">
+		<div class="input-group-prepend">
+			<b-button
+				@click="updateOffset"> 
+				UTC ±
+			</b-button>
+		</div>
+		<input
+			v-bind:value="timezoneString"
+			@input="validate"
+			@change="submitChange"
+			placeholder="[+-]hh:mm"
+			class="form-control" />
+	</div>
+</template>
+
+<script>
+export default {
+	name: 'timezonepicker',
+	model: {
+		prop: 'value',
+		event: 'input'
+	},
+	props: {
+		format: String,
+		value: String,
+		timezonedate: String
+	},
+	data() {
+		return {
+			internal_value: null,
+			initial_value: null,
+			isInitializing: true,
+			inputRegexp: /^[-+\d:]+$/,
+			timezoneRegexp: /^[+-]?(2[0-3]|[0-1]?\d)(:[0-5]?\d|:)?$/
+		}
+	},
+	computed: {
+		timezoneString: {
+			get() {
+				return this.internal_value
+			},
+			set(new_value) {
+				if (this.isInitializing) { return }
+				if (this.timezoneRegexp.test(new_value)) {
+					this.internal_value = new_value
+				}
+			}
+		},
+	},
+	methods: {
+		updateOffset() {
+			// this.date is in day.month.year format
```

Comment:
I have a feeling that there is something dangerous in here since date things are calculated without a library and that always goes wrong eventually..

Reply:
noted and ignored